Improve solution of problem sum of pairs

borgesgfj opened the pull request:
- Overview:
   Here is proposed two new solution to sum of pairs problem, aiming to
   increase the computational task time. Also, is provide a description
   about each of the three solutions proposed.
 - Details:
   - In the file somaDeParesWithOrdenation.js, is performed the
     ordenation of the two arrays and the inner for loop is interrupeted
     when the sum of outter loop element with the inner one exceds the value
     of X
   - In the file somaDeParesWithBinarySearch.js one of two arrays is
     ordered and a binary serach is performed on it aiming to find the
     complementar element of selected by the outter loop (the for loop).

-

borgesgfj committed:
Improve solution of problem.
 - Overview:
   Here is proposed two new solution to sum of pairs problem, aiming to
   increase the computational task time. Also, is provide a description
   about each of the three solutions proposed.
 - Details:
   - In the file somaDeParesWithOrdenation.js, is performed the
     ordenation of the two arrays and the inner for loop is interrupeted
     when the sum of outter loop element with the inner one exceds the value
     of X
   - In the file somaDeParesWithBinarySearch.js one of two arrays is
     ordered and a binary serach is performed on it aiming to find the
     complementar element of selected by the outter loop (the for loop).

-

borgesgfj committed:
solve by using HashMap

cassios commented on soma-de-pares/somaDeParesLinearComplexity.js:
```diff
@@ -0,0 +1,26 @@
+const mapNumbers = new Map();
+const convertArrayToMap = (number, indexInArray) => mapNumbers.set(number, indexInArray);
+function selectNumberPairs(map, array) {
+  let validPairs = [];
```
validPairs pode ser `const` pq o "container" lista não muda.

cassios commented on soma-de-pares/somaDeParesLinearComplexity.js:
```diff
@@ -0,0 +1,26 @@
+const sumResult = 10;
+const numericalOrdering = (a, b) => a - b;
+const mapNumbers = new Map();
+const convertArrayToMap = (number, indexInArray) => mapNumbers.set(number, indexInArray);
```
a função `convertArrayToMap` não está fazendo o que ela diz. Ela está adicionando um item ao mapa.

Esta função deveria receber um array e retornar um mapa. Da forma que está, vai dar problema se vc tiver mais de um testCase pq  mapNumbers é global

borgesgfj replied:
é verdade. Acho melhor eu criar uma função que retorne o mapNumbers .

cassios reviewed (commented):
👏🏽

borgesgfj committed:
Fix function convertArrayToMap

 - The function convertArrayToMap has just one parameter, that is the
   array to be converted in a map. This function returns a map.
 - The variable mapNumbers is definid in the scope of convertArrayToMap
   function instead of the global scope.

cassios commented on soma-de-pares/somaDeParesWithOrdenation.js:
```diff
@@ -0,0 +1,30 @@
+}
+
+for (let n = 0; n < testCase.length; n++) {
+  testCase[n][0].sort(numericalOrdering);
```
a ordenação deveria ficar dentro da `selectNumberPairs` pq ela faz parte do algoritmo.

é necessário ordenar os dois arrays?

borgesgfj replied:
Não é necessário ordenar os dois não. Apenas naquele em que se procura o número complementar

cassios commented on soma-de-pares/somaDeParesLinearComplexity.js:
```diff
@@ -0,0 +1,31 @@
+
+for (let n = 0; n < testCase.length; n++) {
+  console.log(`Caso ${n + 1}: Pares de números cuja soma é ${sumResult}`);
+  console.log(selectNumberPairs(convertArrayToMap(testCase[n][0]), testCase[n][1]));
```
bora passar a conversão para dentro da `selectNumberPairs` pq faz parte do algoritmo

cassios reviewed (commented):
faltou rodar linter em todos os arquivos.

mandou muito bem 👏🏽

borgesgfj committed:
Fix based on PR discusions

 - In somaDeParesWithOrdenation.js just one  array is sorted (the
   arrayNumbers2). This ordering is processed inside the
   selectNumberPairs function.
 - In somaDeParesLinearComplexity.js the conversion of one array to map
   is now performed inside the selectNumberPairs function. The
   selecNumberPairs was changed to receive two array as parameter.
 - Linter was run in all files in this directory to formate the code.

cassios reviewed (approved):
Parabéns 👏🏽  Mandou muitooo bem!!

Lembre-se de juntar os commits antes de mergear
